skynet.go

```diff
@@ -57,33 +132,259 @@ type (
 )
 
 const (
+	// DefaultPortalURL is the default URL of the portal to use.
+	DefaultPortalURL = "https://siasky.net"
+
 	// URISkynetPrefix is the URI prefix for Skynet.
 	URISkynetPrefix = "sia://"
 )
 
 var (
+	// DefaultConnectionOptions contains the default connection options.
+	DefaultConnectionOptions = ConnectionOptions{
+		PortalURL: DefaultPortalURL,
+	}
+
+	// DefaultAddSkykeyOptions contains the default addskykey options.
+	DefaultAddSkykeyOptions = AddSkykeyOptions{
+		ConnectionOptions:   DefaultConnectionOptions,
+		PortalAddSkykeyPath: "/skynet/addskykey",
+	}
+	// DefaultCreateSkykeyOptions contains the default createskykey options.
+	DefaultCreateSkykeyOptions = CreateSkykeyOptions{
+		ConnectionOptions:      DefaultConnectionOptions,
+		PortalCreateSkykeyPath: "/skynet/createskykey",
+	}
+	// DefaultGetSkykeyIDOptions contains the default skykeyid GET options.
+	DefaultGetSkykeyIDOptions = GetSkykeyIDOptions{
+		ConnectionOptions:     DefaultConnectionOptions,
+		PortalGetSkykeyIDPath: "/skynet/skykeyid",
+	}
+	// DefaultGetSkykeyOptions contains the default skykey GET options.
+	DefaultGetSkykeyOptions = GetSkykeyOptions{
+		ConnectionOptions:   DefaultConnectionOptions,
+		PortalGetSkykeyPath: "/skynet/skykey",
+	}
+	// DefaultListSkykeysOptions contains the default skykeys options.
+	DefaultListSkykeysOptions = ListSkykeysOptions{
+		ConnectionOptions:     DefaultConnectionOptions,
+		PortalListSkykeysPath: "/skynet/skykeys",
+	}
+
 	// DefaultDownloadOptions contains the default download options.
 	DefaultDownloadOptions = DownloadOptions{
-		PortalURL:          "https://siasky.net",
+		ConnectionOptions:  DefaultConnectionOptions,
 		PortalDownloadPath: "/",
 	}
 
 	// DefaultUploadOptions contains the default upload options.
 	DefaultUploadOptions = UploadOptions{
-		PortalURL:                    "https://siasky.net",
+		ConnectionOptions:            DefaultConnectionOptions,
```

Comment:
Are we promising any backwards compatibility with the SDKs? Would this potential break that?

Reply:
I guess not if we are just embedding the struct. Mostly wanted to make sure I understand our stance on backwards compatibility for the SDKs

Reply:
We already had some changes that will make backwards compatibility impossible. We definitely will be breaking compatibility with the next version release.